pdftext/pdf/links.py

```diff
@@ -0,0 +1,265 @@
+import ctypes
+import math
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+import pypdfium2 as pdfium
+import pypdfium2.raw as pdfium_c
+
+from pdftext.pdf.utils import matrix_intersection_area
+from pdftext.schema import Bbox, Link, Page, Pages, Span
+
+
+def _get_dest_position(dest) -> Optional[Tuple[float, float]]:
+    has_x = ctypes.c_int()
+    has_y = ctypes.c_int()
+    has_zoom = ctypes.c_int()
+    x_coord = ctypes.c_float()
+    y_coord = ctypes.c_float()
+    zoom_level = ctypes.c_float()
+    success = pdfium_c.FPDFDest_GetLocationInPage(
+        dest,
+        ctypes.byref(has_x),
+        ctypes.byref(has_y),
+        ctypes.byref(has_zoom),
+        ctypes.byref(x_coord),
+        ctypes.byref(y_coord),
+        ctypes.byref(zoom_level)
```

Comment:
nit: you don't need `byref()` here - our ctypesgen-generated bindings are declaring the argtypes, so ctypes should handle this automatically (as per https://docs.python.org/3/library/ctypes.html#type-conversions)

That said, this PR looks great! Though, admittedly this field is a bit above my head.
PS: Feel free to contribute helpers/wrappers to pypdfium2 if you think that would be elegant. e.g. the upcoming v5 branch makes a stab at a `PdfDest` helper (mainly for the bookmarks at this time).

Reply:
Thank you for letting me know! I've removed byref.
Would be happy to contribute, I'll take a look at the v5 branch : )